Examples/FilterDemonstration/Documentation.rst

```diff
@@ -0,0 +1,58 @@
+Filter Comparison
+=================
+
+
+Overview
+--------
+
+There are numerous SimpleITK filters that have similar functions, but very
+important differences. The filters that will be compared are:
+
+ - ``JoinSeriesImageFilter()`` - Joins multiple N-D images into an (N+1)-D  image
+ - ``ComposeImageFilter()`` - Combines several scalar images into a  multicomponent vector image
+ - ``VectorIndexSelectionCastImageFilter()`` - Extracts the selected index of  the input pixel type vector (the input image pixel type must be a vector and  the output a scalar)
+ - ``ExtractImageFilter()`` - Crops an image to the selected region bounds  using vectors; Can also collapse dimensions
```

Comment:
Currently the extract always reduces the image dimension by one unless the dimension is already 2.